tests/core/job/test_job.py

```diff
@@ -236,7 +237,9 @@ def _dispatch(task: Task, job: Job, mode=JobConfig._DEVELOPMENT_MODE):
     _OrchestratorFactory._build_dispatcher()
     _TaskManager._set(task)
     _JobManager._set(job)
-    dispatcher = _StandaloneJobDispatcher(_OrchestratorFactory._orchestrator)
+    dispatcher: Union[_StandaloneJobDispatcher, _DevelopmentJobDispatcher] = _StandaloneJobDispatcher(
```

Comment:
Maybe we can use the parent class instead of the Union. I don't know if mypy is OK with that though.

```suggestion
    dispatcher: JobDispatcher = _StandaloneJobDispatcher(
```

Reply:
I don't think mypy would like that.

Reply:
ok, no problem to keep the Union then.